extensions/wikia/RTE/js/plugins/infobox/plugin.js

```diff
@@ -0,0 +1,153 @@
+					$('.infoboxBuilderDialog').find('.cke_dialog_footer').hide();
+
+					if (infoboxBuilderMarkup) {
+						if (!iframeLoaded) {
+							// needs to be inserted only once, see comment bellow.
+							$('.ckeditor-infobox-builder').html(infoboxBuilderMarkup);
+
+							// this script needs to be executed only once, otherwise ponto stops working on second and
+							// further infobox-builder dialog appearences, which in fact causes blank dialogs without
+							// infobox builder. Because of that, the iFrame html needs to be inserted only once too.
+							loader.processScript(infoboxBuilderScripts);
+							iframeLoaded = true;
+						}
+
+					}
+
+					window.CKEDITOR.on('new-infobox-created', function (event) {
+						var infoboxTitle = event.data;
+
+						CKEDITOR.dialog.getCurrent().hide();
+						RTE.templateEditor.createTemplateEditor(infoboxTitle);
+					});
+				}
+			}
+		});
+
+		CKEDITOR.dialog.getCurrent().hide();
+
+		loader({
+			type: loader.MULTI,
+			resources: {
+				mustache: 'extensions/wikia/PortableInfoboxBuilder/templates/PortableInfoboxBuilderSpecialController_builder.mustache',
+				scripts: 'portable_infobox_builder_js'
+			}
+		}).done(function (assets) {
+			infoboxBuilderMarkup = mustache.render(assets.mustache[0], {
+				iframeUrl: location.origin + '/infobox-builder/',
+				classes: 've-ui-infobox-builder'
+			});
+			infoboxBuilderScripts = assets.scripts;
+
+			RTE.getInstance().openDialog('infoboxBuilder-dialog');
+		});
+	}
+
+	function openInfoboxModal( editor ) {
+		$.get('/api.php?format=json&action=query&list=allinfoboxes&uselang=' + window.wgContentLanguage)
+			.then(function (data) {
+				window.CKEDITOR.dialog.add( 'infobox-dialog', function( editor ) {
+					return {
+						title: $.msg('rte-select-infobox-title'),
+						buttons: [
+						{
+							type : 'button',
+							class: 'infobox-dialog-button',
+							label : '+rte-add-template',
+							onClick : openInfoboxBuilder
+						}
+						],
+						minWidth: 250,
+						minHeight: 300,
+						contents: [
+							{
+								id: 'ckeditorInfoboxPickDialog',
+								label: '',
+								title: '',
+								elements: [
+									{
+										type: 'html',
+										html: getInfoboxListMarkup(data)
+									}
+								]
+							}
+						],
+						onShow: function () {
+							$('.infobox-templates-list').on('click', onInfoboxTemplateChosen);
+						},
+						onHide: function () {
+							$('.infobox-templates-list').off('click', onInfoboxTemplateChosen);
+						}
+					};
+				});
+
+				RTE.getInstance().openDialog('infobox-dialog');
+			});
+	}
+
+	function getInfoboxListMarkup(data) {
+		if (!data || !data.query || !data.query.allinfoboxes || !data.query.allinfoboxes.length) {
+			return '';
+		}
+
+		var markup = '<ul class="infobox-templates-list">';
+
+		data.query.allinfoboxes.forEach(function (infoboxData) {
+			markup += '<li><a data-infobox-name="' + infoboxData.title + '">' + infoboxData.title + '</a></li>';
```

Comment:
Let's escape input here

Reply:
@TK-999 do we need to do it? it is data returned from our API